src/LinuxTracing/ThreadStateManager.h

```diff
@@ -60,10 +62,26 @@ class ThreadStateManager {
 
  private:
   struct OpenState {
-    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns)
-        : state{state}, begin_timestamp_ns{begin_timestamp_ns} {}
+    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns,
+              pid_t wakeup_tid = orbit_base::kInvalidThreadId,
+              pid_t wakeup_pid = orbit_base::kInvalidProcessId,
+              orbit_grpc_protos::ThreadStateSlice::WakeupReason wakeup_reason = orbit_grpc_protos::
+                  ThreadStateSlice::WakeupReason::ThreadStateSlice_WakeupReason_kNotApplicable)
+        : state{state},
+          begin_timestamp_ns{begin_timestamp_ns},
+          wakeup_tid{wakeup_tid},
+          wakeup_pid{wakeup_pid},
+          wakeup_reason{wakeup_reason} {}
     orbit_grpc_protos::ThreadStateSlice::ThreadState state;
     uint64_t begin_timestamp_ns;
+    // The next two fields are optional. We use them to transfer info between interruptible sleep
+    // state (gray) and runnable state (blue), so we can know which tid and pid caused the thread to
```

Comment:
It's not necessarily from interruptible sleep, there are other sleep states. I would also drop the colors as they are a UI details. Also I would simplify the sentence. So in total I would say:

"The next two fields are optional. We use them to indicate which tid and pid caused the thread to transition from a non-runnable to runnable state".